cmd/kueuectl/app/version/version.go

```diff
@@ -0,0 +1,106 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package version
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	"github.com/spf13/cobra"
+	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/cli-runtime/pkg/genericiooptions"
+	k8s "k8s.io/client-go/kubernetes"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+
+	"sigs.k8s.io/kueue/cmd/kueuectl/app/util"
+	"sigs.k8s.io/kueue/pkg/version"
+)
+
+const (
+	kueueNamespace             = "kueue-system"
+	kueueControllerManagerName = "kueue-controller-manager"
+	kueueContainerName         = "manager"
+)
+
+const (
+	versionExample = `  # Prints the client version
+  kueuectl version`
+)
+
+// VersionOptions is a struct to support version command
+type VersionOptions struct {
+	K8sClientset k8s.Interface
+
+	genericiooptions.IOStreams
+}
+
+// NewOptions returns initialized Options
+func NewOptions(streams genericiooptions.IOStreams) *VersionOptions {
+	return &VersionOptions{
+		IOStreams: streams,
+	}
+}
+
+// NewVersionCmd returns a new cobra.Command for fetching version
+func NewVersionCmd(clientGetter util.ClientGetter, streams genericiooptions.IOStreams) *cobra.Command {
+	o := NewOptions(streams)
+
+	cmd := &cobra.Command{
+		Use:     "version",
+		Short:   "Prints the client version",
+		Example: versionExample,
+		Args:    cobra.NoArgs,
+		Run: func(cmd *cobra.Command, args []string) {
+			cobra.CheckErr(o.Complete(clientGetter))
+			cobra.CheckErr(o.Run(cmd.Context()))
+		},
+	}
+
+	return cmd
+}
+
+func (o *VersionOptions) Complete(clientGetter util.ClientGetter) error {
+	var err error
+
+	o.K8sClientset, err = clientGetter.K8sClientset()
+	if err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// Run executes version command
+func (o *VersionOptions) Run(ctx context.Context) error {
+	fmt.Fprintf(o.Out, "Client Version: %s\n", version.GitVersion)
+
+	deployment, err := o.K8sClientset.AppsV1().Deployments(kueueNamespace).Get(ctx, kueueControllerManagerName, v1.GetOptions{})
+	if err != nil {
+		return client.IgnoreNotFound(err)
+	}
+
+	for _, container := range deployment.Spec.Template.Spec.Containers {
+		if container.Name == kueueContainerName {
+			parts := strings.Split(container.Image, ":")
+			fmt.Fprintf(o.Out, "Controller Manager Version: %s\n", parts[1])
```

Comment:
It's not garnteed to have one and only one ":" in the image name (it could be 0, 1 and 2 at least ), the tag is optional and the registry url can contain a port. 

https://kubernetes.io/docs/concepts/containers/images/#image-names

Take this into account.

What we can do:

1. Always print the full image
2. Print `parts[len(parts)-1]`

1 It's better in my opinion